e2e/test/scenarios/admin/database-routing2.cy.spec.ts

```diff
@@ -0,0 +1,306 @@
+const { H } = cy;
+import {
+  QA_DB_CREDENTIALS,
+  QA_POSTGRES_PORT,
+  USER_GROUPS,
+} from "e2e/support/cypress_data";
+import type { DatabaseData, User } from "metabase-types/api";
+
+import { interceptPerformanceRoutes } from "./performance/helpers/e2e-performance-helpers";
+import {
+  durationRadioButton,
+  openSidebarCacheStrategyForm,
+  saveCacheStrategyForm,
+} from "./performance/helpers/e2e-strategy-form-helpers";
+
+const { ALL_USERS_GROUP, DATA_GROUP, COLLECTION_GROUP } = USER_GROUPS;
+
+const personA = {
+  first_name: "Don",
+  last_name: "RouterA",
+  email: "routerA@metabase.test",
+  password: "12341234",
+  login_attributes: {
+    destination_database: "postgres_two",
+  },
+  user_group_memberships: [
+    { id: ALL_USERS_GROUP, is_group_manager: false },
+    { id: COLLECTION_GROUP, is_group_manager: false },
+    { id: DATA_GROUP, is_group_manager: false },
+  ],
+};
+
+const personB = {
+  first_name: "Tom",
+  last_name: "RouterB",
+  email: "routerB@metabase.test",
+  password: "12341234",
+  login_attributes: {
+    destination_database: "postgres_three",
+  },
+  user_group_memberships: [
+    { id: ALL_USERS_GROUP, is_group_manager: false },
+    { id: COLLECTION_GROUP, is_group_manager: false },
+    { id: DATA_GROUP, is_group_manager: false },
+  ],
+};
+
+describe("admin > database > database routing", { tags: ["@external"] }, () => {
+  before(() => {
+    cy.signInAsAdmin();
+    H.queryWritableDB(
+      "SELECT datname from pg_database WHERE datname = 'lead'",
+    ).then((res: { rows: any[] }) => {
+      if (res.rows.length === 0) {
+        H.queryWritableDB("CREATE DATABASE lead;");
+      }
+    });
+
+    H.queryWritableDB(
+      "SELECT datname from pg_database WHERE datname = 'postgres_two'",
+    ).then((res: { rows: any[] }) => {
+      if (res.rows.length === 0) {
+        H.queryWritableDB("CREATE DATABASE postgres_two;");
+      }
+    });
+
+    H.queryWritableDB(
+      "SELECT datname from pg_database WHERE datname = 'postgres_three'",
+    ).then((res) => {
+      if (res.rows.length === 0) {
+        H.queryWritableDB("CREATE DATABASE postgres_three;");
+      }
+    });
+
+    const leadConfig = {
```

Comment:
I created a third postgres db instead of using writable as the primary db because I was having issues with the writable db schema getting wiped. I think it was something to do with being restored/reset, but I couldn't figure it out. The primary db schema is used for routing to the destination dbs, so it wasn't working without it